### PR TITLE
Github workflow for running PSI-cardinality installation and benchmark tests for every commit/PR

### PR DESCRIPTION
**Pull request by rutujasurve94**
Adding tests for running psi-cardinality installation and benchmark tests for push on GitHub

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    - '**.cpp'
+  pull_request:
+    types: [ opened, synchronize, reopened ]
+    paths:
+    - '**.cpp'
+
+jobs:
+  PSI-cpp-tests:
+
+    runs-on: ubuntu-latest
```
Comment (bcebere) [inline .github/workflows/tests.yml:16]: Maybe we should the tests on MacOS(Windows?) too.

Comment (schoppmp) [reply .github/workflows/tests.yml:16]: I'm not sure the C++ code even compiles on Windows. It should be possible to get it to compile, but I don't think it's a priority for now.

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        echo "deb [arch=amd64] https://storage.googleapis.com/bazel-apt stable jdk1.8" | sudo tee /etc/apt/sources.list.d/bazel.list
+        curl https://bazel.build/bazel-release.pub.gpg | sudo apt-key add -
+        # Step 3: Install and update Bazel
+        sudo apt-get update && sudo apt-get install bazel
+        sudo apt-get install --only-upgrade bazel
+
+    - name: Lint with clang-format
+      run: |
+        # lint files (all .cpp and .h files) inplace
+        find ./psi-cardinality/  \( -iname *.h -o -iname *.cpp \) -a -not -path "*go*" | xargs clang-format -i -style='file'
```
[the remark this answers is not in the document]
Comment (schoppmp) [reply .github/workflows/tests.yml:36]: @bcebere Isn't the style enforced by the .clang-format file? If we pass -style="google", why do we need .clang-format?

[the remark this answers is not in the document]
Comment (bcebere) [reply .github/workflows/tests.yml:36]: passing -style="file" overwrites the .clang-format preferences.
I took the config from another openmined project, it's derived from google style, but it has some customizations.

We can drop the config, apparently it's not used now anyway.

[the remark this answers is not in the document]
Comment (rutujasurve94) [reply .github/workflows/tests.yml]: sure, done

File: .github/workflows/tests.yml (hunk 1 above)
Comment (rutujasurve94) [reply .github/workflows/tests.yml]: done, added ubuntu and macos

**Pull request merged**